spec/inputs/check_boxes_input_spec.rb

```diff
@@ -358,6 +358,31 @@
 
   end
 
+  describe ':collection for a has_and_belongs_to_many association' do
+
+    before do
+      @output_buffer = ''
+      mock_everything
+
+      concat(semantic_form_for(@freds_post) do |builder|
+        concat(builder.input(:authors, as: :check_boxes, collection: Author.all))
+      end)
+    end
+
+    it 'should render checkboxes' do
+      # I'm aware these two lines test the same thing
```

Comment:
I wish I had a better answer, but I am a bit ignorant as to this testing methodology. Sorry :(